### DISCUSSION TITLE
How can I redirect a "new user" (performed the first singin) to a different URL?

### DISCUSSION THREAD

**Discussion by gaborvecsei** [nextauthjs/next-auth#9125] [Help]
Let's say my users can sign-in with Google. I would like to run a user onboarding once it's their first time signing in.
I could easily do this if I could define the redirection in the `signIn` callback (e.g.: `return "/profile/onboarding"`) and singnal a successful signin.
But afaik this is not possible, only for unsuccessful signings.

What is the go-to solution for this problem?

**Comment by melisciar** [comment]
Have you tried adding a newUser page on the configuration?
```
pages: {
    ...
    newUser: '/profile/onboarding' // New users will be directed here on first sign in (leave the property out if not of interest)
  }
```
https://next-auth.js.org/configuration/pages
